xql/src/xql/apply.py

```diff
@@ -356,34 +307,39 @@ def filter_records(ds: xr.Dataset, query: str) -> dd.DataFrame:
         query (str): The query string for filtering the dataset.
 
     Returns:
-        dd.DataFrame: A dask DataFrame containing the filtered records.
+        pd.DataFrame: A pandas DataFrame containing the filtered records.
     """
 
     # Parse the query expression
     expr = parse_one(query)
 
-    # Find Limit and Offset clauses in the query
+    # Find Limit, Offset and OrderBy clauses in the query
+    orderby_clause = expr.find(exp.Order)
     limit_clause = expr.find(exp.Limit)
     offset_clause = expr.find(exp.Offset)
 
     # Convert xarray Dataset to dask DataFrame
-    ddf = ds.to_dask_dataframe()
+    df = ds.to_dataframe()
+
+    # Apply orderby clause if present
+    if orderby_clause:
+        df = apply_orderby(orderby_clause, df)
 
     # Initialize start location for slicing
     start_loc = 0
 
     # Apply offset clause if present
     if offset_clause:
         start_loc = int(offset_clause.expression.args['this'])
-        ddf = ddf.loc[start_loc:]
+        df = df.iloc[start_loc:]
 
     # Apply limit clause if present
     if limit_clause:
         limit = int(limit_clause.expression.args['this'])
-        ddf = ddf.loc[:start_loc + limit - 1]
+        df = df.iloc[:start_loc + limit]
 
     # Compute and return the filtered DataFrame
-    return ddf.compute()
+    return df.reset_index()
```

Comment:
not too sure here, but shouldn't this be done at L323 (right after the `df = ds.to_dataframe()` call)?

Reply:
We can call this anywhere.